react/__tests__/DeviceCoordinates.test.tsx

```diff
@@ -0,0 +1,164 @@
+      })
+
+      Object.defineProperty(navigator, 'permissions', {
+        get() {
+          return { query: mockedPermissionsQuery }
+        },
+      })
+
+      const { getCurrentPosition } = navigator.geolocation
+      const mockedGetCurrentPosition = getCurrentPosition as jest.MockedFunction<
+        typeof getCurrentPosition
+      >
+
+      mockedGetCurrentPosition.mockImplementation(success => {
+        callOrder.push('geolocation')
+
+        return Promise.resolve(
+          success({
+            coords: {
+              latitude: vtexOfficeAddress.geoCoordinates[0],
+              longitude: vtexOfficeAddress.geoCoordinates[1],
+            },
+          } as GeolocationPosition)
+        )
+      })
+
+      Object.defineProperty(navigator, 'geolocation', {
+        get() {
+          return { getCurrentPosition: mockedGetCurrentPosition }
+        },
+      })
+
+      const mockedOnSuccess = jest.fn()
+
+      renderComponent({
+        deviceCoordinatesProps: { onSuccess: mockedOnSuccess },
+        graphqlMocks: [reverseGeocode],
+      })
+
+      const button = screen.getByRole('button')
+
+      fireEvent.click(button)
+
+      expect(mockedPermissionsQuery).toHaveBeenCalledTimes(1)
+      expect(mockedGetCurrentPosition).toHaveBeenCalledTimes(1)
+      expect(callOrder).toEqual(['permissions', 'geolocation'])
+      await waitFor(() => {
+        expect(mockedOnSuccess).toHaveBeenCalledTimes(1)
+      })
+    })
+
+    it("should get an address from coordinates on a browser that doesn't support navigator.permissions", async () => {
+      Object.defineProperty(navigator, 'permissions', {
+        get() {
+          return undefined
+        },
+      })
+
+      const { getCurrentPosition } = navigator.geolocation
+      const mockedGetCurrentPosition = getCurrentPosition as jest.MockedFunction<
+        typeof getCurrentPosition
+      >
+
+      mockedGetCurrentPosition.mockImplementation(success =>
+        Promise.resolve(
+          success({
+            coords: {
+              latitude: vtexOfficeAddress.geoCoordinates[0],
+              longitude: vtexOfficeAddress.geoCoordinates[1],
+            },
+          } as GeolocationPosition)
+        )
+      )
+
+      Object.defineProperty(navigator, 'geolocation', {
+        get() {
+          return { getCurrentPosition: mockedGetCurrentPosition }
+        },
+      })
+
+      const mockedOnSuccess = jest.fn()
+
+      renderComponent({
+        deviceCoordinatesProps: { onSuccess: mockedOnSuccess },
+        graphqlMocks: [reverseGeocode],
+      })
+
+      const button = screen.getByRole('button')
+
+      fireEvent.click(button)
+
+      expect(mockedGetCurrentPosition).toHaveBeenCalledTimes(1)
+      await waitFor(() => {
+        expect(mockedOnSuccess).toHaveBeenCalledTimes(1)
+      })
+    })
+  })
+
+  // TODO
+  describe('when geolocation is blocked', () => {})
+
+  // TODO
+  describe('when geolocation is prompt', () => {})
+
+  afterEach(() => {
```

Comment:
can you move this up to the beginning of the function?

Reply:
Sure. Do you know if there is any eslint rule for that kind of thing?

Reply:
I don't think so :-/